lazyflow/utility/io_util/OMEZarrStore.py

```diff
@@ -256,13 +257,42 @@ def _get_zarr_cache_max_size() -> int:
     return math.floor(caches_max * permissible_fraction_max)
 
 
-def _unescape_and_get_store(uri: str, mode="r", **kwargs) -> FSStore:
+def _ensure_connection_and_get_store(uri: str, mode="r", **kwargs) -> FSStore:
     if uri.startswith("file:"):
         # Zarr's FSStore implementation doesn't unescape file URLs before piping them to
         # the file system. We do it here the same way as in pathHelpers.uri_to_Path.
         # Primarily this is to deal with spaces in Windows paths (encoded as %20).
         uri = os.fsdecode(unquote_to_bytes(uri))
-    return FSStore(uri, mode=mode, **kwargs)
+    store = FSStore(uri, mode=mode, **kwargs)
+    try:
+        store["test"]
+        # Any error not handled here is either a successful connection (404 for "test"), or an unknown problem
+    except KeyError as e:
+        # FSStore wraps some errors in KeyError (FileNotFoundError even double-wrapped)
+        if isinstance(e.__context__, ClientConnectorError):
+            raise ConnectionError(f"Could not connect to {e.__context__.host}:{e.__context__.port}.") from e
+    except ClientResponseError as e:
+        if not (e.status == 403 and uri.startswith("https://")):
+            raise
+        # 403 indicates we need authentication; FSStore maps https: to HTTPFileSystem, which cannot authenticate
+        split_bucket = uri.split("/", 4)
+        if len(split_bucket) < 5:  # Does not follow S3 pattern (https://s3.server.org/bucket/file)
+            raise ConnectionError(
+                f"Server refused permission to read {uri}.\nIt seems authentication is required, but ilastik does not support this kind of server yet."
+            ) from e
+        base_uri_inc_bucket = "/".join(split_bucket[:4])
+        sub_uri = split_bucket[4]
+        fs = s3fs.S3FileSystem(anon=False, endpoint_url=base_uri_inc_bucket)
+        store = FSStore(sub_uri, fs=fs, **kwargs)
+        try:
+            store["test"]
+        except KeyError as ke:
+            if isinstance(ke.__context__, PermissionError):
+                # This probably is an S3-compatible store, but auth rejected/not found.
+                raise ConnectionError(
+                    f"Server refused permission to read {uri}.\nCheck your S3 credentials setup."
```

Comment:
Same here, a hint on the s3 credential setup might be good (or is it so heterogeneous that it is not possible to hint at something?)

Reply:
There are three different ways they might be authenticating with their bucket (creds file, env variables, token). It's a bit too impractical for ilastik to determine which way it is here just for a nicer error. Plus, all of those would have been manually set up by the user, so they should know how they are doing it (or who set it up for them)...